pkg/operator/apiserver/controller/apiservice/apigroup.go

```diff
@@ -9,62 +9,56 @@ import (
 
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	kubeinformers "k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/rest"
 	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
-func newEndpointPrecondition(kubeInformers kubeinformers.SharedInformerFactory) func(apiServices []*apiregistrationv1.APIService) (bool, error) {
-	// this is outside the func so it always registers before the informers start
-	endpointsLister := kubeInformers.Core().V1().Endpoints().Lister()
-
+func checkEndpointsPresence(endpointsLister corev1listers.EndpointsLister, apiServices []*apiregistrationv1.APIService) (bool, error) {
```

Comment:
this function should match exactly the previous one, i have just changed the signature.